console/client/src/features/modules/ModuleDetails.tsx

```diff
@@ -1,15 +1,30 @@
-import { useContext } from 'react'
-import { useLocation, useNavigate, useSearchParams } from 'react-router-dom'
+import React from 'react'
 import { SelectedModuleContext } from '../../providers/selected-module-provider'
-import { TabSearchParams, TabType, TabsContext } from '../../providers/tabs-provider'
-import { textColor } from '../../utils/style.utils'
+import { TabType, TabsContext, TabSearchParams } from '../../providers/tabs-provider'
+import { textColor, urlSearchParamsToObject } from '../../utils'
+import { useSearchParams } from 'react-router-dom'
+import { modulesContext } from '../../providers/modules-provider'
 
 export function ModuleDetails() {
-  const { selectedModule } = useContext(SelectedModuleContext)
-  const { tabs, setTabs, setActiveTab } = useContext(TabsContext)
-  const navigate = useNavigate()
-  const location = useLocation()
-  const [ searchParams ] = useSearchParams()
+  const modules = React.useContext(modulesContext)
+  const { selectedModule, setSelectedModule } = React.useContext(SelectedModuleContext)
+  const { tabs, setTabs, setActiveTab } = React.useContext(TabsContext)
+  const [ searchParams, setSearchParams ] = useSearchParams()
+  const moduleId = searchParams.get('module')
+  //P1 when mounting with a valid module in query params set selected module
+  React.useEffect(() => {
+    if(moduleId) {
+      const module = modules.modules.find(module => module?.name === moduleId)
+      module && setSelectedModule(module)
+    }
+  })
+  //P2 when query param module changes update selected module if valid
+  React.useEffect(() => {
```

Comment:
Do you prefer the `React.useEffect` over just `useEffect` here? Is this something we can lint on? I wonder if there's a standard approach.

Reply:
React isn't a ESM package destructuring doesn't do anything. Wether you import useEffect or React.useEffect you are getting the entire react library.  I use to destructure but stoped at Pluralsight when a an architect called this out and we moved to a no n destructuring pattern for react which is either UMD | CJS but never ESM.